File: src/openfl/display/Sprite.nim
import ./DisplayObjectContainer

type
  Sprite* = ref object of DisplayObjectContainer
    buttonMode*: bool
    useHandCursor*: bool

proc initSprite*(self: Sprite) =
  self.initDisplayObjectContainer()
  self.buttonMode = false
  self.useHandCursor = false

proc newSprite*(): Sprite =
  let self = Sprite()
  self.initSprite()
  return self

File: kfl/src/main/kotlin/openfl/display/Sprite.kt
package openfl.display

import openfl.display._internal.IBitmapDrawableType
import openfl.geom.Matrix
import openfl.geom.Rectangle

open class Sprite : DisplayObjectContainer() {

    var buttonMode: Boolean = false
    var useHandCursor: Boolean = true

    // Graphics object placeholder
    // val graphics: Graphics by lazy { Graphics(this) }

    init {
        __drawableType = IBitmapDrawableType.SPRITE
    }

    override fun __getBounds(rect: Rectangle, matrix: Matrix) {
        super.__getBounds(rect, matrix)
        // Add graphics bounds here when implemented
    }
}
